astroid/brain/brain_dataclasses.py

```diff
@@ -231,9 +232,11 @@ def infer_dataclass_attribute(
 
 
 def infer_dataclass_field_call(
-    node: AssignName, ctx: context.InferenceContext = None
-) -> Generator:
+    node: Call, ctx: Optional[context.InferenceContext] = None
+) -> Optional[Generator]:
```

Comment:
```suggestion
) -> Generator:
```
The return type will then only be `Generator`.